src/Form/index.tsx

```diff
@@ -406,6 +417,80 @@ function Form({
   const hasRedirected = useRef<boolean>(false);
   const elementClicks = useRef<any>({}).current;
 
+  // Reference for handling the button state after navigating to the next step
+  const nextActionStateRef = useRef<NextActionStateRef>({
+    // This is for the most recently clicked Next Step button.
+    latestClickedButton: null,
+
+    // This is for the getNewStep state.
+    // isGettingNewStep = true
+    // await getNewStep()
+    // isGettingNewStep = false
+    isGettingNewStep: false,
+    timerIdGettingNewStep: undefined,
+
+    // This is for the buttonOnClick state.
+    // Since buttonOnClick is invoked externally by other components,
+    // we cannot use await for state handling like getNewStep() in this component scope.
+    // Instead, inside buttonOnClick, set isNextButtonAction to TRUE at the initial entry point
+    // and set it to FALSE just before the function returns to manage the buttonOnClick state.
+    isNextButtonAction: false,
+    timerIdNextActionFlag: undefined
+  });
+
+  const setNextButtonActionFlag = (
```

Comment:
could we pull these out into a separate utility file? this component is already quite long